### DISCUSSION TITLE
Migration/Backup&Restore Advice

### DISCUSSION THREAD

**Discussion by ndanyluk** [nextcloud/helm#391] [Q&A] [answered]
Hi all, I'm currently running NC in a VM since many components seemed to "want" to be more and more in control of the OS itself (and I didn't want to figure out how to set up NC on-top of a HA k8s cluster.

My question is: to migrate successfully to a Nextcloud-Helm install, would it be easiest to
A) new install using the chart, then restore from backup within the new install?
B) chart install but pointing to my data on disk and letting it figure it out?

I feel like A is safer, but this is something of an edge case I imagine

**Comment by provokateurin** [comment, marked as answer]
I think either way is good. It depends on what you want to have later. Since the data will be on a pvc anyway, there is probably not a lof ot benefits to the second one, other than having a separate (physical I assume?) disk for your data.

**Reply by ndanyluk** [reply]
Ok good to know.  So the data is currently on a ZFS dataset and this is (currently and for at least a while) a single node k3s cluster.  My plan was to simply mount either a hostPath or NFS share into the container.  My backup has been tested and has full /var/www/html/ (I think, need to double check but it covers everything in NC's backup guide) and a dump of my MariaDB Nextcloud db.

My main fear with B was that NC might attempt to overwrite or mangle the data since it's a fresh install that doesn't know about the prior install

**Reply by provokateurin** [reply]
If you make sure your data folder is in the right place there will be no problem with B.

Before upgrading you should make sure that the versions match, then it's less likely problems will happen.

**Reply by ndanyluk** [reply]
Awesome thank you! Just to confirm, btw, since I didn't see it in the docs - is backup/restore achieved just like the regular nextcloud docs suggest? i.e. maintenance mode, copy folders, dump DB, remove maintenance mode?

**Reply by provokateurin** [reply]
Yes!

**Reply by provokateurin** [reply]
Make sure to have the correct permissions on the folders.
